internal/pkg/template/templates/workloads/partials/cf/workload-container.yml

```diff
@@ -29,17 +29,14 @@
 {{- if eq .WorkloadType "Load Balanced Web Service"}}
   PortMappings:
     - ContainerPort: !Ref ContainerPort
-{{- if .NLB}}
-  {{if ne .NLB.Listener.TargetPort .NLB.MainContainerPort}} {{/*No need to add additional port if the target port is the same as image port*/}}
-    - ContainerPort: {{.NLB.Listener.TargetPort}}
-      {{- if eq .NLB.Listener.Protocol "UDP" }}
-      Protocol: udp
-      {{- else }}
-      Protocol: tcp
-      {{- end }}
+{{- if .NLB}} {{ $nlbListener := .NLB.Listener }} 
+  {{- if and (eq $nlbListener.TargetContainer .WorkloadName) (ne $nlbListener.TargetPort .NLB.MainContainerPort)}}
```

Comment:
:ack: this looks good, what about this feedback from the issue:

> It appears that the cf partial below is not sidecar aware, and assumes that the nlb target is always the main workload container.

To me it sounded like the NLB target group is pointing to the wrong port?

Reply:
With a manifest like this:
```
nlb:
  port: 80/tcp
  target_container: envoy

...

sidecars:
  envoy:
    # envoy admin port
    port: 9901
    image: public.ecr.aws/hashicorp/envoy-alpine:latest
```
The NLB target group's target port will be 9901 instead of 80. 

My understanding for the quoted comment was specifically for the code:
https://github.com/aws/copilot-cli/blob/2ac6b01cdeacc0a0fb81ffdcaf731ba2bfc4cb82/internal/pkg/template/templates/workloads/partials/cf/workload-container.yml#L33

The (bugged) code assume the target container is main container, and add the target port to main container's port mapping if it's not the same as the container port. It fails to take into consideration the possibility that the target port being different could be due to the target port is read from a sidecar.

Reply:
Thanks for the explanation, totally makes sense!